pkg/apis/eventsource/v1alpha1/types.go

```diff
@@ -0,0 +1,410 @@
+/*
+Copyright 2018 BlackRock, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"github.com/argoproj/argo-events/gateways/common"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/apimachinery/pkg/watch"
+)
+
+// SourceType is the type of event source
+type SourceType string
+
+// Possible values for event source
+// These must match the field names from `Source`
+const (
+	SourceTypeWebhook     SourceType = "webbook"
```

Comment:
not sure this is a good idea.. see above comment. this makes it very hard to generalize this object/API. with these specific types, one can't create an anonymous event source without breaking the API!